aws-rds-optiongroup/src/main/java/software/amazon/rds/optiongroup/ListHandler.java

```diff
@@ -0,0 +1,40 @@
+package software.amazon.rds.optiongroup;
+
+import java.util.stream.Collectors;
+
+import software.amazon.awssdk.services.rds.RdsClient;
+import software.amazon.awssdk.services.rds.model.DescribeOptionGroupsResponse;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.OperationStatus;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+public class ListHandler extends BaseHandlerStd {
+
+    @Override
+    public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<RdsClient> proxyClient,
+            final Logger logger) {
+
+        final DescribeOptionGroupsResponse describeOptionGroupsResponse =
+                proxy.injectCredentialsAndInvokeV2(Translator.describeOptionGroupsRequest(request.getNextToken()),
+                        proxyClient.client()::describeOptionGroups);
+
```

Comment:
since you are not using call-chain, I think you should wrap this in `try-catch` and map exceptions manually.

*Note: you should not use call-chain for list handler as it can cause memory overflow, bc there might be an increased amount of resources under the account and listing all of them, memoizing into callback might kill lambda container

Reply:
Fixed. Thank you!